packages/protocol/contracts/governance/BlockchainParameters.sol

```diff
@@ -26,12 +20,10 @@ contract BlockchainParameters is Ownable, Initializable, UsingPrecompiles {
     uint256 nextValueActivationEpoch;
   }
 
-  ClientVersion private minimumClientVersion;
   uint256 public blockGasLimit;
```

Comment:
Are we going to deploy BlockchainParameters contract from scratch? Since this change will break proxy design pattern if we will only update implementation

Reply:
I'm open to any approach and I have not been involved with any celo contract deployments, yet. The only change I really need is the removal of the `should exit when minimum version is updated` test. So if you have a recommendation which changes to leave out or move to a separate PR, I'll be happy to adjust this PR.

Reply:
Proxy design pattern in solidity works in a way that it is not possible to change variable order, delete any existing variable or add any variable between already existing variables. The only thing that can be done is to append new variable at the end of variable declaration. If this is not followed during proxy upgrade, storage of the whole proxy will be broken. Please note that it is possible to change name of the variable. There is more info about it in following `[article](https://medium.com/coinmonks/proxy-pattern-and-upgradeable-smart-contracts-45d68d6f15da)`.

For control of this mechanism during PR merge validation we have implemented version control check. Anytime this check requires you to change Major version it means there is breaking change in storage layout. In such case it is not possible update the contract but only deploy new one (with need to update all other dependent contracts) 

Rather than deleting variable you can mark is as deprecated with example [here](https://github.com/celo-org/celo-monorepo/blob/db1130f612f55a1f22d8e3d9a245f2bf264c047c/packages/protocol/contracts/governance/Governance.sol#LL51C18-L51C18)

If you would have any questions we can have a call a go through it :-)

Reply:
That makes sense. I was expecting something like this but I was not sure about the details.